WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/WooShipping Create Shipping Labels/WooShippingCreateLabelsViewModel.swift

```diff
@@ -211,6 +230,34 @@ final class WooShippingCreateLabelsViewModel: ObservableObject {
         self.destinationAddressStatus = .verified
         self.onLabelPurchase = nil
         self.stores = stores
+        Task {
+            await loadRequiredData()
+        }
```

Comment:
I checked whether we need to load the required data for an existing shipping label. While testing, I noticed that we allow editing a shipping label's origin and destination addresses. This shouldn't be allowed. I logged an issue under M3 to address this.  https://github.com/woocommerce/woocommerce-ios/issues/15268

Considering the above, we should avoid loading the origin addresses for an existing shipping label. I think we can work on it in this PR. 

What do you think?

Reply:
Good catch about editing addresses on existing shipping labels!

Regarding fetching required data, I think it's still needed for store settings as with the changes in this PR, we no longer set the cached settings by default. The settings are still loaded from the remote and cached results are only used when the remote request fails.

I updated the logic check in `loadRequiredData` to check for existing origin address and only fetch more options if it's not available (b68ee49ac0b281d26e365f1ba661dc751ff9bd93). Do you think this works? @selanthiraiyan